### DISCUSSION TITLE
Adding extra regressors to the model

### DISCUSSION THREAD

**Discussion by jamal-54** [ourownstory/neural_prophet#258] [Q&A - get help using NeuralProphet]
I am trying to forecast load with this neuralprophet. with historical data, I want to add temperature as an input parameter to my forecasting features. Is it possible to implement that with this neuralprophet? If yes, can you suggest to me how to do that?

**Comment by ourownstory** [comment]
Hi @jamal-54 
If temperature is a measurement (not known of the future), you can use a [lagged-regressor](https://neuralprophet.com/html/lagged-regressors.html)
if it is a known forecast, you can use a simple [future-regressor](https://neuralprophet.com/html/future-regressors.html?highlight=future%20regressor)

**Reply by Msaleh87** [reply]
I tried these links, but they weren't working, I Googled the new links:
Modelling Lagged Regressors: https://neuralprophet.com/html/lagged-regressors.html

Modelling Future Regressors: https://neuralprophet.com/html/future-regressors.html?highlight=future%20regressor

**Reply by ourownstory** [reply]
Thank you @Msaleh87, I updated the links.

**Reply by Msaleh87** [reply]
@ourownstory 
So I prepared my data as shown below:
![Screenshot 2022-10-03 165322](https://user-images.githubusercontent.com/42920113/193680423-88e3a373-898d-41fd-a557-bd5f244ee402.png)

This is my code, L12 is the dataset you see in the pic above:

```
L12M = L_prophet_5min_T[0:(12*30*24*12)] # the dataset show in the above pic
m = NeuralProphet(n_forecasts=(2*24*12),
    n_lags=(16*24*12),
    yearly_seasonality="auto",
    weekly_seasonality="auto",  
    daily_seasonality="auto",
    growth="off", seasonality_reg= 0.5, optimizer = 'AdamW',
    learning_rate = 0.1, n_changepoints="auto", changepoints_range = "auto",  batch_size = 16 ) 

m.add_country_holidays("US", mode="additive") 
m = m.add_future_regressor(name='A')
m = m.add_future_regressor(name='B')
m = m.add_future_regressor(name='C')
metrics = m.fit(L12M, freq="5min") 
m.highlight_nth_step_ahead_of_each_forecast(step_number=m.n_forecasts)

# the future values of A, B, and C
future_regressors_df = pd.DataFrame(data={'A': L_prophet_5min_T['A'][(12*30*24*12):(12*30*24*12)+576], 
                                          'B': L_prophet_5min_T['B'][(12*30*24*12):(12*30*24*12)+576],
                                          'C': L_prophet_5min_T['C'][(12*30*24*12):(12*30*24*12)+576]})
```
The model trains and saves fine, then when I try to forecast using this code, I get an error:

```
P_horizon = 576
future = m.make_future_dataframe(df = L12M, periods = P_horizon, n_historic_predictions=True) 
forecast1 = m.predict(df=future)
```
![Screenshot 2022-10-03 170128](https://user-images.githubusercontent.com/42920113/193681742-af08552c-5df4-4586-9a2c-b411ac3c076c.png)

Should I add an argument for the future regressors inside the forecast?!
